src/realm/object-store/impl/object_notifier.hpp

```diff
@@ -22,22 +22,24 @@
 #include <realm/object-store/impl/collection_notifier.hpp>
 
 #include <realm/keys.hpp>
+#include <realm/table.hpp>
 
 namespace realm {
 
 namespace _impl {
 class ObjectNotifier : public CollectionNotifier {
 public:
-    ObjectNotifier(std::shared_ptr<Realm> realm, TableKey table, ObjKey obj);
+    ObjectNotifier(std::shared_ptr<Realm> realm, Obj obj);
 
 private:
-    TableKey m_table;
-    ObjKey m_obj;
+    Obj m_obj;
```

Comment:
We can't hang on to the Obj unfortunately because it has the same problem as holding on to a Table. You will have to keep storing the ObjKey and TableKey as before because those are safe across threads. Take a closer look at the ListNotifier constructor, even though it takes a List it doesn't hold on to it from that point, it only stores it's keys.

Reply:
Gotcha!

I implemented `do_attach_to()` similar to the `ListNotifier` to attach to the correct `Transaction`: 128b48d.